### PR TITLE
[torch_glow] Add support for ListUnpack and Split  ops

### PR DESCRIPTION
**Pull request by dangus36**
Summary:
Added support for split and ListUnpack ops in PyTorchModelLoader.

Test Plan:
Added multiple unit tests in split_test.py.

### CODE REVIEW

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 1]
```diff
@@ -2476,6 +2478,96 @@ Error PyTorchModelLoader::loadListConstruct(const torch::jit::Node *ptNode) {
   RETURN_ERR(addValueMapping(outputs[0], std::move(glowIVal)));
 }
 
+Error PyTorchModelLoader::loadListUnpack(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:2484]: RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 1, outputs, -1));
Please check this before you call inputs[0].

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -2476,6 +2478,96 @@ Error PyTorchModelLoader::loadListConstruct(const torch::jit::Node *ptNode) {
...
+  return Error::success();
+}
+
+Error PyTorchModelLoader::loadSplit(const torch::jit::Node *ptNode) {
+  auto inputs = ptNode->inputs();
+  auto outputs = ptNode->outputs();
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 3, outputs, -1));
+
+  glow::NodeValue input;
+  if (hasGlowNodeValueForValue(inputs[0])) {
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:2505]: Please add explicit name for each input, like, inputs[SplitInput::input].
You can find examples in other ops.

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 3]
```diff
@@ -2476,6 +2478,96 @@ Error PyTorchModelLoader::loadListConstruct(const torch::jit::Node *ptNode) {
...
+    // possible.
+    int64_t splitInt;
+    ASSIGN_VALUE_OR_RETURN_ERR(splitInt, iValToInt(splitIVal));
+
+    const auto axisDim = input.dims()[axis];
+    if (splitInt < 1) {
+      return MAKE_ERR(
+          "[Split] Single `split` integer value must be greater than zero.");
+    } else if (splitInt > axisDim) {
+      return MAKE_ERR("[Split] Single `split` integer value must be less "
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:2535]: Is this neccessary?

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 4]
```diff
@@ -2476,6 +2478,96 @@ Error PyTorchModelLoader::loadListConstruct(const torch::jit::Node *ptNode) {
...
+                      "than input_dims[axis]");
+    }
+    auto splitDimT = static_cast<dim_t>(splitInt);
+
+    dim_t quotient = axisDim / splitDimT; // Calculate best split size
+    dim_t remainder =
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:2541]: remainder is a reserved word, please use another name.

**Review by zrphercule — CHANGES_REQUESTED** [review]
Overall LGTM, please address all the comments and we will be able to land it.
Thanks!

**Comment by zrphercule** [comment]
Also, please resolve the conflict and rebase it. It seems the conflict is not a big issue. Thanks!

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 1 above)
Comment (dangus36) [reply torch_glow/src/PyTorchModelLoader.cpp:2493]: Added in newest commit

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 2 above)
Comment (dangus36) [reply torch_glow/src/PyTorchModelLoader.cpp:2516]: Added in newest commit

**Comment by zrphercule** [comment]
@dangus36 My apologies this pr along with some other prs are automatically closed during Xmas + new year holidays. I reopened them all, and could you update them a little bit? Thanks!

**Comment by dangus36** [comment]
@zrphercule No worries, will rebase and upload now.

**Pull request closed without merging**